packages/onboarding-ui/src/forms/CreateCloudWorkspaceForm/CreateCloudWorkspaceForm.stories.tsx

```diff
@@ -6,6 +6,21 @@ import WorkspaceUrlInput from './WorkspaceUrlInput';
 
 type Args = ComponentProps<typeof CreateCloudWorkspaceForm>;
 
+const isValidDomainName = async (domainName: string) => {
```

Comment:
I think that makes sense to turn this validation into a list of validations, and iterate on that validating the clauses and returning true or false,  maybe creating a helper or component for todo the validations in this way.

Reply:
Nice, how about this next commit?